public/calendar_reboot.js

```diff
@@ -50,35 +50,52 @@ function CheckIt() {
     }.bind(this));
    
    
-    //WHEN PAGE LOADS
-    //first load the allCalendarIds from storage.
-    var allCalendarIds = this.store.getAllCalendarIds() || {};
-   
-    //GOING THROUGH THE KEYS OF THE DICTIONARY allCalendarIds
-    for (var key in allCalendarIds) {
-      if (allCalendarIds.hasOwnProperty(key)) {
-        this.addCalendarToDropdown(key, allCalendarIds[key]);
-      }
-    }
+    // WHEN PAGE LOADS
+    // Load the calendar Ids from storage and fill the dropdown with calendar
+    // titles.
+    this.store.getAllCalendarIds()
+        .then(function (allCalendarIds) {
+            // Add calendar titles to dropdown.
+            
+            for (var key in allCalendarIds) {
+                if (allCalendarIds.hasOwnProperty(key)) {
+                this.addCalendarToDropdown(key, allCalendarIds[key]);
+                }
+            }
+        }.bind(this))
+        
```

Comment:
It would be best not to put whitespace randomly everywhere because it tricks the reader into thinking things are unrelated. In this case, `.catch()` is being called on the return value of `.then()`, but it kind of looks like it stands alone somehow.